crates/biome_js_analyze/src/lint/complexity/no_extra_boolean_cast.rs

```diff
@@ -188,7 +192,36 @@ impl Rule for NoExtraBooleanCast {
             ExtraBooleanCastType::DoubleNegation => "Remove redundant double-negation",
             ExtraBooleanCastType::BooleanCall => "Remove redundant `Boolean` call",
         };
-        mutation.replace_node(node.clone(), node_to_replace.clone());
+
+        // Check if the Boolean call is inside a unary negation and the argument needs parentheses
+        let replacement = if matches!(extra_boolean_cast_type, ExtraBooleanCastType::BooleanCall) {
+            // Check if this Boolean call is inside a unary negation
+            if let Some(unary_expr) = node.syntax().parent().and_then(JsUnaryExpression::cast) {
+                if matches!(unary_expr.operator(), Ok(JsUnaryOperator::LogicalNot)) {
+                    // Check if the argument is a complex expression that needs parentheses
+                    if needs_parentheses_when_negated(node_to_replace) {
+                        // Wrap in parentheses to preserve operator precedence
+                        AnyJsExpression::JsParenthesizedExpression(
+                            make::js_parenthesized_expression(
+                                make::token(T!['(']),
+                                node_to_replace.clone(),
+                                make::token(T![')']),
+                            ),
+                        )
+                    } else {
+                        node_to_replace.clone()
+                    }
+                } else {
+                    node_to_replace.clone()
+                }
+            } else {
+                node_to_replace.clone()
+            }
+        } else {
+            node_to_replace.clone()
+        };
```

Comment:
And you can definitely refactor this so that these elses aren't necessary.

Something like this probably:
```rust
let mut replacement = ...;

if cond {
    replacement = AnyJsExpression::JsParenthesizedExpression(
                            make::js_parenthesized_expression(
                                make::token(T!['(']),
                                replacement,
                                make::token(T![')']),
                            ),
                        )
}

mutation.replace_node(node.clone(), replacement);